test: `test_nested_dtypes` fails with Polars==0.20.30

spotted in the random ci versions job in https://github.com/narwhals-dev/narwhals/pull/1220

```
_______________________________________________________________________________ test_nested_dtypes ________________________________________________________________________________

    @pytest.mark.skipif(
        parse_version(pd.__version__) < parse_version("2.2.0"),
        reason="too old for pyarrow types",
    )
    def test_nested_dtypes() -> None:
        df = pl.DataFrame(
            {"a": [[1, 2]], "b": [[1, 2]], "c": [{"a": 1}]},
            schema_overrides={"b": pl.Array(pl.Int64, 2)},
        ).to_pandas(use_pyarrow_extension_array=True)
        nwdf = nw.from_native(df)
        assert nwdf.schema == {
            "a": nw.List(nw.Int64),
            "b": nw.Array(nw.Int64, 2),
            "c": nw.Struct({"a": nw.Int64}),
        }
        df = pl.DataFrame(
            {"a": [[1, 2]], "b": [[1, 2]], "c": [{"a": 1}]},
            schema_overrides={"b": pl.Array(pl.Int64, 2)},
        )
        nwdf = nw.from_native(df)
>       assert nwdf.schema == {
            "a": nw.List(nw.Int64),
            "b": nw.Array(nw.Int64, 2),
            "c": nw.Struct({"a": nw.Int64}),
        }

tests/frame/schema_test.py:235: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
narwhals/dataframe.py:1128: in schema
    return super().schema
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ┌───────────────────────────────────────┐
| Narwhals DataFrame                    |
| Use `.to_native` to see native output |
└───────────────────────────────────────┘

    @property
    def schema(self) -> Schema:
>       return Schema(self._compliant_frame.schema.items())
E       AttributeError: 'function' object has no attribute 'items'

narwhals/dataframe.py:89: AttributeError
```

--- a/narwhals/_polars/dataframe.py
+++ b/narwhals/_polars/dataframe.py
@@ -68,6 +68,12 @@ class PolarsDataFrame:
     def __getattr__(self, attr: str) -> Any:
         if attr == "collect":  # pragma: no cover
             raise AttributeError
+        if attr == "schema":
+            schema = self._native_frame.schema
+            return {
+                name: native_to_narwhals_dtype(dtype, self._dtypes)
+                for name, dtype in schema.items()
+            }
 
         def func(*args: Any, **kwargs: Any) -> Any:
             args, kwargs = extract_args_kwargs(args, kwargs)  # type: ignore[assignment]
@@ -84,14 +90,6 @@ class PolarsDataFrame:
         if self._backend_version < (0, 20, 28):  # pragma: no cover
             return self._native_frame.__array__(dtype)
         return self._native_frame.__array__(dtype)
-
-    @property
-    def schema(self) -> dict[str, Any]:
-        schema = self._native_frame.schema
-        return {
-            name: native_to_narwhals_dtype(dtype, self._dtypes)
-            for name, dtype in schema.items()
-        }
 
     def collect_schema(self) -> dict[str, Any]:
         if self._backend_version < (1,):  # pragma: no cover

--- a/narwhals/_polars/utils.py
+++ b/narwhals/_polars/utils.py
@@ -84,7 +84,7 @@ def native_to_narwhals_dtype(dtype: Any, dtypes: DTypes) -> DType:
     if dtype == pl.List:
         return dtypes.List(native_to_narwhals_dtype(dtype.inner, dtypes))
     if dtype == pl.Array:
-        if parse_version(pl.__version__) < (1, 0):  # pragma: no cover
+        if parse_version(pl.__version__) < (0, 20, 30):  # pragma: no cover
             return dtypes.Array(
                 native_to_narwhals_dtype(dtype.inner, dtypes), dtype.width
             )

